js/page-specific/spay-neuter-calculator.js

```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
+			scrollTop: $spayNeuterFormResults.offset().top
+		}, 1000);			
+	};
 
-			callback = callback || function() {};
+	/**
+	 * Sets the visibility of the target, and fires a callback when done.
+	 */
+	var setVisibility = function ($target, shouldHide, callback) {
+		var ANIMATION_DURATION = 300;
+		var visibilityFilterSelector = shouldHide ? ':hidden' : ':visible';
+		var $formControlWrapper = $target.closest('.bc-form-control');
 
-			if ($formControlWrapper.is(visibilityFilterSelector)) {
-				callback();
-				return;
-			}
+		callback = callback || function() {};
 
-			$formControlWrapper.attr('aria-hidden', shouldHide);
+		if ($formControlWrapper.is(visibilityFilterSelector)) {
+			callback();
+			return;
+		}
 
-			if (shouldHide) 
-				$formControlWrapper.fadeOut(ANIMATION_DURATION, callback);
-			else
-				$formControlWrapper.fadeIn(ANIMATION_DURATION, callback);				
-		},
+		$formControlWrapper.attr('aria-hidden', shouldHide);
 
-		/**
-		 * Clears all hidden form elements, so they're not already filled out when they're revealed. 
-		 * This is really just a safeguard for anyone who gets click-happy.
-		 */
-		clearEverythingOnTheFormAfterThis = function ($target) {
-			var targetIndex = $spayNeuterFormInputElements.index($target);
+		if (shouldHide) 
+			$formControlWrapper.fadeOut(ANIMATION_DURATION, callback);
+		else
+			$formControlWrapper.fadeIn(ANIMATION_DURATION, callback);				
+	};
 
-			$spayNeuterFormInputElements.map(function (index, item) {
-				var $item = $(item);
+	/**
+	 * Clears all hidden form elements, so they're not already filled out when they're revealed. 
+	 * This is really just a safeguard for anyone who gets click-happy.
+	 */
+	var clearEverythingOnTheFormAfterThis = function ($target) {
+		var targetIndex = $spayNeuterFormInputElements.index($target);
 
-				if ($spayNeuterFormInputElements.index(item) > targetIndex)
-					$item.val('').removeAttr('checked');
-			});
-		},
+		$spayNeuterFormInputElements.map(function (index, item) {
+			var $item = $(item);
 
-		/**
-		 * Event binding.
-		 */
-		init = function () {
-			var YES_RADIO_INDEX = 0,
-				NO_RADIO_INDEX = 1;
-
-			$residentField.eq(YES_RADIO_INDEX).on('click', function (event) {
-				clearEverythingOnTheFormAfterThis(event.target);
-				setVisibility($spayNeuterFormResults, true, function () {
-					setVisibility($spayNeuterFormButton, true, function () {
-						setVisibility($zipCodeField, true, function () {
-							setVisibility($catPitBullField, true, function () {
-								setVisibility($publicAssistanceField, false);
-							});
+			if ($spayNeuterFormInputElements.index(item) > targetIndex)
+				$item.val('').removeAttr('checked');
+		});
+	};
+
+	/**
+	 * Event binding.
+	 */
+	var init = function () {
+		var YES_RADIO_INDEX = 0;
+		var NO_RADIO_INDEX = 1;
+
+		$residentField.eq(YES_RADIO_INDEX).on('click', function (event) {
+			clearEverythingOnTheFormAfterThis(event.target);
+			setVisibility($spayNeuterFormResults, true, function () {
```

Comment:
I think we've talked about this in the past, but we could probably refactor this some way. Maybe using an array of objects, or something like that.